read_batteryCheck.c

```diff
@@ -0,0 +1,20 @@
+#include "checker.h"
+
+void readtemperatureorsoc(float temperatureorsoc,int min_val,int max_val)
+{
+    if((temperatureorsoc < min_val) || (temperatureorsoc > max_val)) 
+            printf("Temperature/State of Charge/Charge Rate out of range!\n");
+ }
+
+float ConvertToCelcius(float farenheit)
+{
+    return (farenheit - 32) * 5 / 9;
+}
+
+void warningalert_SOC(int temperatureorsoc)
+{
+    if(temperatureorsoc <=SOC_MIN+4) 
```

Comment:
hard coded number 4 could lead to maintenance issues. there will be doubts on why 4 was used here.
Using constants usually will remove those doubts.